nvdaHelper/vbufBase/backend.cpp

```diff
@@ -249,3 +259,56 @@ VBufBackend_t::~VBufBackend_t() {
 	LOG_DEBUG(L"base Backend destructor called"); 
 	nhAssert(runningBackends.count(this) == 0);
 }
+
+VBufStorage_controlFieldNode_t* VBufBackend_t::reuseExistingNodeInRender(VBufStorage_controlFieldNode_t* parent, VBufStorage_fieldNode_t* previous, int docHandle, int ID) {
+	LOG_DEBUG(L"Try to reuse node with docHandle "<<docHandle<<L", and ID "<<ID);
+	if(parent->alwaysRerenderDescendants||parent->alwaysRerenderChildren) {
+		LOG_DEBUG(L"Won't  find a node to reuse as parent says always rerender children");
+		return nullptr;
+	}
+	// Locate a possible existing node with the given docHandle and ID
+	auto existingNode=this->getControlFieldNodeWithIdentifier(docHandle,ID);
+	if(!existingNode) {
+		LOG_DEBUG(L"Could not locate a node with docHandle "<<docHandle<<L", and ID "<<ID);
+		return nullptr;
+	}
+	// Ensure the node allows us to reuse it
+	if(!existingNode->allowReuseInAncestorUpdate) {
+		LOG_DEBUG(L"Existing node refuses to be reused");
+		return nullptr;
+	}
+	if(existingNode->denyReuseIfPreviousSiblingsChanged) {
+		VBufStorage_controlFieldNode_t* previousControlFieldNode=nullptr;
+		for(auto tempNode=previous;tempNode!=nullptr;tempNode=tempNode->getPrevious()) {
+			previousControlFieldNode=dynamic_cast<VBufStorage_controlFieldNode_t*>(tempNode);
+			if(previousControlFieldNode) break;
+		}
+		VBufStorage_referenceNode_t* previousReferenceNode=dynamic_cast<VBufStorage_referenceNode_t*>(previousControlFieldNode);
+		if(previousControlFieldNode&&!previousReferenceNode) {
+			LOG_DEBUG(L"Previous controlFieldNode was not a referenceNode");

```

Comment:
A comment here would be nice explaining that if the previous ControlFieldNode isn't a reference node, that means it is a newly added node.